File: Visual Studio Product Team/ADO.NET Entity Framework/[VB]-ADO.NET Entity Framework/VB/EmployeeTracker/ViewModel/EmailViewModel.vb
' Copyright © Microsoft Corporation.  All Rights Reserved.
' This code released under the terms of the 
' Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)



Imports Microsoft.VisualBasic
Imports System
Imports EmployeeTracker.Model

Namespace EmployeeTracker.ViewModel

    ''' <summary>
    ''' ViewModel of an individual Email
    ''' </summary>
    Public Class EmailViewModel
        Inherits ContactDetailViewModel
        ''' <summary>
        ''' The Email object backing this ViewModel
        ''' </summary>
        Private email As Email

        ''' <summary>
        ''' Initializes a new instance of the EmailViewModel class.
        ''' </summary>
        ''' <param name="detail">The underlying Email this ViewModel is to be based on</param>
        Public Sub New(ByVal detail As Email)
            If detail Is Nothing Then
                Throw New ArgumentNullException("detail")
            End If

            Me.email = detail
        End Sub

        ''' <summary>
        ''' Gets the underlying Email this ViewModel is based on
        ''' </summary>
        Public Overrides ReadOnly Property Model() As ContactDetail
            Get
                Return Me.email
            End Get
        End Property

        ''' <summary>
        ''' Gets or sets the actual email address
        ''' </summary>
        Public Property Address() As String
            Get
                Return Me.email.Address
            End Get

            Set(ByVal value As String)
                Me.email.Address = value
                Me.OnPropertyChanged("Address")
            End Set
        End Property
    End Class
End Namespace

File: Visual Studio Product Team/Official Visual Studio 2010 Samples for C# 4.0/46872-Official Visual Studio 2010 Samples for C# 4.0/ADO.NET Entity Framework/C#/EmployeeTracker/ViewModel/EmailViewModel.cs
// Copyright © Microsoft Corporation.  All Rights Reserved.
// This code released under the terms of the 
// Microsoft Public License (MS-PL, http://opensource.org/licenses/ms-pl.html.)


namespace EmployeeTracker.ViewModel
{
    using System;
    using EmployeeTracker.Model;

    /// <summary>
    /// ViewModel of an individual Email
    /// </summary>
    public class EmailViewModel : ContactDetailViewModel
    {
        /// <summary>
        /// The Email object backing this ViewModel
        /// </summary>
        private Email email;

        /// <summary>
        /// Initializes a new instance of the EmailViewModel class.
        /// </summary>
        /// <param name="detail">The underlying Email this ViewModel is to be based on</param>
        public EmailViewModel(Email detail)
        {
            if (detail == null)
            {
                throw new ArgumentNullException("detail");
            }

            this.email = detail;
        }

        /// <summary>
        /// Gets the underlying Email this ViewModel is based on
        /// </summary>
        public override ContactDetail Model
        {
            get { return this.email; }
        }

        /// <summary>
        /// Gets or sets the actual email address
        /// </summary>
        public string Address
        {
            get
            {
                return this.email.Address;
            }

            set
            {
                this.email.Address = value;
                this.OnPropertyChanged("Address");
            }
        }
    }
}
